brush-parser/src/parser.rs

```diff
@@ -255,7 +255,7 @@ peg::parser! {
         rule bang() -> bool = specific_word("!") { true }
 
         rule pipe_sequence() -> Vec<ast::Command> =
-            c:command() ++ (specific_operator("|") linebreak()) { c }
+            c:command() ++ ((non_posix_extensions_enabled() specific_operator("|&")) / specific_operator("|") linebreak()) { c }
```

Comment:
Maybe there's something I'm missing, but doesn't this rule now accept `x | y` or `x |& y` yet discard the difference between them?

Reply:
This rule acts as a separator between commands. |& actually peeks without consuming inside the command. Is there a better approach maybe?

Reply:
My understanding is that the `++` combinator _does_ consume the indicated separator (`|&` or `|` in this case). So this might be the right place put this operator, but we need to capture the fact that it was used instead of `|`.

From https://docs.rs/peg/latest/peg/:

> `expression ++ delim` - Delimited repeat (one or more): match one or more repetitions of expression delimited with delim and return the results as a Vec.